clap_derive/src/utils/doc_comments.rs

```diff
@@ -117,10 +83,200 @@ fn is_blank(s: &str) -> bool {
+                    writer.indentation += tab_width;
+                }
+                Event::Start(Tag::Emphasis) => writer.start_style(style_emphasis),
+                Event::Start(Tag::Strong) => writer.start_style(style_strong),
+                Event::Start(Tag::Strikethrough) => writer.start_style(style_strike_through),
+                Event::Start(Tag::Link { dest_url, .. }) => {
+                    writer.start_link(dest_url);
+                    writer.start_style(style_link);
+                }
+
+                Event::End(TagEnd::Paragraph) => {
+                    if short.is_none() {
+                        short = Some(writer.output.trim().to_owned());
+                    }
+                    writer.endline();
+                    writer.newline();
+                }
+                Event::End(TagEnd::Heading(..)) => {
+                    writer.end_style(style_heading);
+                    writer.endline();
+                    writer.newline();
+                }
+                Event::End(TagEnd::List(_)) => {
+                    let list = list_indices.pop();
+                    debug_assert!(list.is_some());
+                    if list_indices.is_empty() {
+                        writer.newline();
+                    }
+                }
+                Event::End(TagEnd::Item) => {
+                    writer.indentation -= tab_width;
+                    writer.endline();
+                }
+                Event::End(TagEnd::Emphasis) => writer.end_style(style_emphasis),
+                Event::End(TagEnd::Strong) => writer.end_style(style_strong),
+                Event::End(TagEnd::Strikethrough) => writer.end_style(style_strike_through),
+                Event::End(TagEnd::Link) => {
+                    writer.end_link();
+                    writer.end_style(style_link);
+                }
+                Event::End(
+                    TagEnd::Image | TagEnd::BlockQuote(_) | TagEnd::HtmlBlock | TagEnd::CodeBlock,
+                ) => { /* IGNORED */ }
+
+                Event::Text(segment) => write!(writer, "{segment}"),
+                Event::Code(code) => {
+                    writer.start_style(style_code);
+                    write!(writer, "{code}");
+                    writer.end_style(style_code);
+                }
+                // There is not really anything useful to do with block level html.
+                Event::Html(html) => write!(writer, "{html}"),
+                // At some point we could support custom tags like `<red>`
+                Event::InlineHtml(html) => write!(writer, "{html}"),
+                Event::SoftBreak => write!(writer, " "),
+                Event::HardBreak => writer.endline(),
+                // TODO for anything useful we'd need to know the terminal width
+                Event::Rule => {
+                    writer.endline();
+                    writer.newline();
+                    write!(writer, "---\n\n");
+                }
```

Comment:
Should we render rule as `---` or is there anything else we can do without terminal width